src/bridge.cpp

```diff
@@ -111,45 +112,43 @@ double bridgeAngle(
 
     if (support_threshold > 0 && (supported_regions.area() / (skin_outline.area() + 1)) < support_threshold)
     {
-        Polygons bb_poly;
-        bb_poly.add(boundary_box.toPolygon());
+        Shape bb_poly;
+        bb_poly.push_back(boundary_box.toPolygon());
 
         // airBelow is the region below the skin that is not supported, it extends well past the boundary of the skin.
         // It needs to be shrunk slightly so that the vertices of the skin polygon that would otherwise fall exactly on
         // the air boundary do appear to be supported
 
         const coord_t bb_max_dim = std::max(boundary_box.max_.X - boundary_box.min_.X, boundary_box.max_.Y - boundary_box.min_.Y);
-        const Polygons air_below(bb_poly.offset(bb_max_dim).difference(prev_layer_outline).offset(-10));
+        const Shape air_below(bb_poly.offset(bb_max_dim).difference(prev_layer_outline).offset(-10));
 
-        Polygons skin_perimeter_lines;
-        for (ConstPolygonRef poly : skin_outline)
+        OpenLinesSet skin_perimeter_lines;
+        for (const Polygon& poly : skin_outline)
         {
-            if (poly.empty())
-                continue;
-            skin_perimeter_lines.add(poly);
-            skin_perimeter_lines.back().emplace_back(poly.front());
+            if (! poly.empty())
+            {
+                skin_perimeter_lines.emplace_back(poly.toPseudoOpenPolyline());
+            }
         }
 
-        Polygons skin_perimeter_lines_over_air(air_below.intersectionPolyLines(skin_perimeter_lines));
+        OpenLinesSet skin_perimeter_lines_over_air(air_below.intersection(skin_perimeter_lines));
 
         if (skin_perimeter_lines_over_air.size())
         {
             // one or more edges of the skin region are unsupported, determine the longest
             coord_t max_dist2 = 0;
             double line_angle = -1;
-            for (PolygonRef air_line : skin_perimeter_lines_over_air)
+            for (const OpenPolyline& air_line : skin_perimeter_lines_over_air)
             {
-                Point2LL p0 = air_line[0];
-                for (unsigned i = 1; i < air_line.size(); ++i)
+                for (auto iterator = air_line.beginSegments(); iterator != air_line.endSegments(); ++iterator)
                 {
-                    const Point2LL& p1(air_line[i]);
-                    coord_t dist2 = vSize2(p0 - p1);
+                    const Point2LL vector = (*iterator).start - (*iterator).end;
```

Comment:
I think I tried that and couldn't make it work for some reason :thinking: but I may be wrong. If you can add the operator, that's perfect, but if something doesn't work, don't push too hard :smile:

Reply:
Ok, now I remember :smile: `operator*` returns a `Segment` object which is a transitory structure. However `operator->` requires a pointer as return type. But as the segment object is build inside the operator, you can't return a pointer to it. So that is not possible as far as I understand.